viseron/components/yolo/object_detector.py

```diff
@@ -0,0 +1,122 @@
+"""YOLO object detector."""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+from ultralytics import YOLO
+
+from viseron import Viseron
+from viseron.domains.object_detector import AbstractObjectDetector
+from viseron.domains.object_detector.detected_object import DetectedObject
+
+from .const import (
+    COMPONENT,
+    CONFIG_DEVICE,
+    CONFIG_HALF_PRECISION,
+    CONFIG_IOU,
+    CONFIG_MIN_CONFIDENCE,
+    CONFIG_MODEL_PATH,
+    CONFIG_OBJECT_DETECTOR,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+def setup(vis: Viseron, config, identifier) -> bool:
+    """Set up the YOLO object_detector domain."""
+    ObjectDetector(vis, config, identifier)
+
+    return True
+
+
+class ObjectDetector(AbstractObjectDetector):
+    """YOLO object detection."""
+
+    def __init__(self, vis: Viseron, config, camera_identifier) -> None:
+        super().__init__(
+            vis, COMPONENT, config[CONFIG_OBJECT_DETECTOR], camera_identifier
+        )
+
+        try:
```

Comment:
In this try/except it would be good to raise the error `DomainNotReady`, which would set the component/domain in an error state and retries will be made.

The `edgetpu` component does something similar.

Reply:
Also, you can ignore my comment in the other thread about return `False` from setup, i was overthinking things